atlas/AtlasMl/atlasml/ml/MLPipelines/PipelineWorkflows.py

```
@@ -0,0 +1,226 @@
+        medoids = np.array(entry["vector"] for entry in clusters)
+        similarity_scores = np.array([compute_cosine_similarity(embedding, medoid) for medoid in medoids])
+        best_medoid_idx = int(np.argmax(similarity_scores))
+        if best_medoid_idx == -1: return
+        if similarity_scores[best_medoid_idx] < 0.5: return
+        competency = self.weaviate_client.get_embeddings_by_property(CollectionNames.COMPETENCY.value, "cluster_id", clusters[best_medoid_idx]["properties"]["cluster_id"])
+        if not competency: return
+        competency_to_match = competency[0]
+
+        properties = {
+                    "text_id": uuid,
+                    "text": text ,
```

Flake8 (7.2.0)

[error] whitespace before ','

(E203)